New libwebsocket breaks FreeBSD compilation

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
Any after #14888 merge.

**OS/device including version:**
FreeBSD 11.1

**Issue description:**
After the inclusion of the libwebsocket library as thirdparty, compilation in FreeBSD throws errors:

```
[ 52%] Compiling ==> thirdparty/lws/client/client-parser.c
In file included from thirdparty/lws/client/client.c:22:
thirdparty/lws/private-libwebsockets.h:239:10: fatal error: 'sys/prctl.h' file
      not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
[ 52%] In file included from thirdparty/lws/client/client-handshake.c:1:
thirdparty/lws/private-libwebsockets.h:239:10: fatal error: 'sys/prctl.h' file
      not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
[ 52%] Compiling ==> thirdparty/lws/client/ssl-client.c
In file included from thirdparty/lws/client/client-parser.c:22:
thirdparty/lws/private-libwebsockets.h:239:10: fatal error: 'sys/prctl.h' file
      not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
In file included from thirdparty/lws/client/ssl-client.c:22:
thirdparty/lws/private-libwebsockets.h:239:10: fatal error: 'sys/prctl.h' file
      not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
1 error generated.
scons: *** [thirdparty/lws/client/client.x11.tools.64.llvm.o] Error 1
1 error generated.
scons: *** [thirdparty/lws/client/client-handshake.x11.tools.64.llvm.o] Error 1
1 error generated.
scons: *** [thirdparty/lws/client/client-parser.x11.tools.64.llvm.o] Error 1
1 error generated.
scons: *** [thirdparty/lws/client/ssl-client.x11.tools.64.llvm.o] Error 1
scons: building terminated because of errors.
```

The library is crossplatform, but BSD are not correctly determined as only differentiate for OSX and iPhone to not include the offending file.

**Steps to reproduce:**

```
$ uname
FreeBSD
$ git pull --rebase upstream master
From https://github.com/godotengine/godot
 * branch                master     -> FETCH_HEAD
Already up to date.
Current branch master is up to date.
$ scons platform=x11 use_llvm=yes -j 4
```

--- a/thirdparty/lws/lws_config_private.h
+++ b/thirdparty/lws/lws_config_private.h
@@ -81,7 +81,7 @@
 
 /* Define to 1 if you have the <sys/prctl.h> header file. */
 #define LWS_HAVE_SYS_PRCTL_H
-#if defined(OSX_ENABLED) || defined(IPHONE_ENABLED)
+#if defined(OSX_ENABLED) || defined(IPHONE_ENABLED) || defined(__FreeBSD__) || defined(__OpenBSD__)
 #undef LWS_HAVE_SYS_PRCTL_H
 #endif
 

